clang/lib/Sema/SPIRVBuiltins.td

```diff
@@ -726,10 +728,12 @@ foreach IType = [UChar, UShort, UInt, ULong] in {
   }
 }
 
-foreach IType = [Char, Short, Int, Long] in {
+foreach IType = [TrueChar, Short, Int, Long] in {
```

Comment:
changing https://github.com/intel/llvm/pull/1463/files#diff-35388f048c1dc54db0a9491da9317c69R290 to `char` is the actual fix I believe. This change will trigger unwanted implicit conversions.

`Char` is bound to `std::int8_t` (always signed char) while `TrueChar` is the actual `char` type which can be unsigned. I added `TrueChar` as a hack for the `__spirv_ocl_printf` but I don't think it should be kept.

Reply:
Ok

Reply:
Sorry the link don't seem to work properly, this line https://github.com/intel/llvm/blob/e875fbf334bd10612637d5d04574b369485d2a53/clang/lib/Sema/SPIRVBuiltins.td#L290 needs to changed from `schar` to `char` so the function name can be built correctly (with https://github.com/intel/llvm/pull/1463/commits/e875fbf334bd10612637d5d04574b369485d2a53 it is now post-fixing with `_Rschar`)

Reply:
Ok